Client/multiplayer_sa/CMultiplayerSA_Weapons.cpp

```diff
@@ -200,6 +200,35 @@ void _declspec(naked) HOOK_Fx_AddBulletImpact()
     }
 }
 
+//////////////////////////////////////////////////////////////////////////////////////////
+//
+// CVisibilityPlugins::RenderWeaponPedsForPC
+// 
+// Fix for the bright objects after weapon change sometimes
+// 
+//////////////////////////////////////////////////////////////////////////////////////////
+#define HOOKPOS_CVisibilityPlugins_RenderWeaponPedsForPC 0x733123
+#define HOOKSIZE_CVisibilityPlugins_RenderWeaponPedsForPC 5
+static constexpr DWORD CONTINUE_CVisibilityPlugins_RenderWeaponPedsForPC = 0x733128;
+static void _declspec(naked) HOOK_CVisibilityPlugins_RenderWeaponPedsForPC()
+{
+    _asm
+    {
+        mov eax, 5DF4E0h
+        call eax // call CPed::ResetGunFlashAlpha
+
+        mov eax, 5533B0h
+        mov ecx, ebx
+
+        pushad
+        push 0
+        call eax // call CPed::RemoveLighting
+        popad
```

Comment:
pushad and popad are unnecessary here

Reply:
You're right; that's a leftover from my crash debugging